src/components/Picker/AutoSuggestText.tsx

```diff
@@ -1,61 +1,43 @@
 
 import * as Autosuggest from 'react-autosuggest';
 import * as React from 'react';
-import Card from './Card';
 import * as style from './Picker.scss';
 import Chip from '../Chip';
 import { themr, ThemedComponentClass } from 'react-css-themr';
 import { TEXT_FIELD } from '../ThemeIdentifiers';
-
 import * as baseTheme from './TextField.scss';
-
 export interface Props {
-  itemsList?: object[];
   theme?: any;
   placeholder?: string;
   autoSuggestMethods: any;
   stateProps: any;
```

Comment:
stateProps and autoSuggestMethods types need to be defined.  `:any` is bad, and I am getting code smell with "state" props here as well.  In general, it's better practice to have flat hierarchy in your props, vs. a nested structure. THat leaves everyone guessing, or browsing around the definition to figure out that hierarchy.

Reply:
@harsimranb Agree! but some of the methods in stateprops conflicts with other props and I receive a couple of warnings when I tried to change the method names. so as of now I implemented the interface and defined stateprops so functionality wise it works perfectly without changing too much of code.